src/Model/Scope/Condition.php

```diff
@@ -350,7 +350,7 @@ protected function valueToWords(Model $model, $value): string
         }
 
         // use the referenced model title if such exists
-        if ($field && ($field->reference ?? false) && $field->reference->getOwner()->isEntity()) {
+        if ($field && ($field->reference ?? false)) {
             // make sure we set the value in the Model parent of the reference
             // it should be same class as $model but $model might be a clone
             $field->reference->getOwner()->set($field->short_name, $value);
```

Comment:
@georgehristov it is not ok to mutate Model from any scope code. What is the correct way to fix this code, eg. clone the reference incl. the parent? On cloned model, we can then call `createEntity()` easily and fix the test.